src/components/BottomSheet/BottomSheet.jsx

```diff
@@ -0,0 +1,140 @@
+  const [bottomSpacerHeight, setBottomSpacerHeight] = useState(0)
+
+  const styles = useMemo(() => makeStyles({ isTopPosition }), [isTopPosition])
+
+  useEffect(() => {
+    const maxHeight = toolbarNode
+      ? window.innerHeight - toolbarNode.offsetHeight
+      : window.innerHeight - 1
+
+    const mediumHeight = Math.round(maxHeight * 0.33)
+    const actionButtonsHeight = parseFloat(
+      getComputedStyle(headerContentRef.current).getPropertyValue('height')
+    )
+    const actionButtonsBottomMargin = parseFloat(
+      getComputedStyle(headerContentRef.current).getPropertyValue(
+        'padding-bottom'
+      )
+    )
+    const minHeight =
+      headerRef.current.offsetHeight +
+      actionButtonsHeight +
+      actionButtonsBottomMargin
+
+    // Used so that the bottomSheet can be opened to the top,
+    // without stopping at the content height
+    const bottomSpacerHeight = maxHeight - innerContentRef.current.offsetHeight
+
+    setPeekHeights([minHeight, mediumHeight, maxHeight])
+    setInitPos(mediumHeight)
+    setBottomSpacerHeight(bottomSpacerHeight)
+  }, [innerContentRef, headerContentRef, toolbarNode])
+
+  const handleOnIndexChange = snapIndex => {
+    const maxHeightSnapIndex = peekHeights.length - 1
+
+    if (snapIndex === maxHeightSnapIndex && !isTopPosition) {
+      setIsTopPosition(true)
+    }
+    if (snapIndex !== maxHeightSnapIndex && isTopPosition) {
+      setIsTopPosition(false)
+    }
+  }
+
+  return (
+    <MuiBottomSheet
+      peekHeights={peekHeights}
+      defaultHeight={initPos}
+      backdrop={false}
+      fullHeight={false}
+      onIndexChange={snapIndex => handleOnIndexChange(snapIndex)}
+      styles={{ root: styles.root }}
+      threshold={0}
+      // springConfig doc : https://www.react-spring.io/docs/hooks/api
+      springConfig={{
+        tension: defaultBottomSheetSpringConfig.tension,
+        friction: defaultBottomSheetSpringConfig.friction,
+        clamp: defaultBottomSheetSpringConfig.clamp
+      }}
+      disabledClosing={true}
+      snapPointSeekerMode="next"
+    >
+      <div ref={innerContentRef}>
+        <div
+          data-testid="bottomSheet-header"
+          className="u-w-100 u-h-2-half u-pos-relative u-flex u-flex-items-center u-flex-justify-center"
+          ref={headerRef}
+        >
+          <div style={styles.indicator} />
+        </div>
+        <Stack
+          style={styles.stack}
+          className="u-flex u-flex-column u-ov-hidden"
+          spacing="s"
+        >
+          <Paper
+            ref={headerContentRef}
+            style={styles.header}
+            className="u-h-3 u-flex u-flex-items-center u-pb-half"
+            elevation={0}
+            square
+          >
+            {header}
+          </Paper>
+          {content}
+        </Stack>
+      </div>
+      <div style={{ height: bottomSpacerHeight }} />
+    </MuiBottomSheet>
+  )
+}
+
+export default React.memo(BottomSheet)
```

Comment:
Can't we reuse the component that is already present in UI ? https://github.com/cozy/cozy-ui/blob/5e82e627b408639c5f028d7201c007dd6a843c13/react/Viewer/Footer/BottomSheetWrapper.jsx

Reply:
> A mon avis non il vaut mieux faire l'inverse, remplacer celui de cozy-ui par celui de CCO2 qui est je pense mieux construit et plus facilement réutilisable

In retrospect we should have taken out the one from viewer and put it at the root of the UI and worked on it rather than making a new component here again and saying we'll have to put it back in later. We'll do four time the work :'(

Reply:
Oui ça revient au même, il n'y a pas eu de temps perdu :) J'ai juste fait un copier/coller pour travailler dans l'app, sans lien avec cozy-ui, parce qu'on a des délais à tenir. Le composant ici a été pensé pour être directement sortable dans cozy-ui, il faudra ensuite refactor un peu le Viewer pour qu'il utilise ce nouveau composant de cozy-ui.